### PR TITLE
added a new Streamlit UI for better readability of users and updated requirements.txt

### PR DESCRIPTION
**Pull request by debrupf2946**
While using Gradio the text display is short and its difficult to read and comprehend 
This new streamlit look will definitely improve UX of usesers
Screenshot 2024-02-29 at 1 55 06 PM

### CODE REVIEW

**Comment by kmehant** [comment]
Looks good, is there a way we can host this? like done for the gradio?

**Comment by kmehant** [comment]
https://github.com/c2siorg/Project-Explainer/pull/19#issuecomment-1970799243

Would you like to host it? Thank you.

File: project_explainer_ui/streamlit_ui.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from gh_explainer import Explainer
+
+def summarize(summarization_type, github_project_url, github_project_branch="main", huggingface_model_id="gpt2"):
+    gptExplainer = Explainer(huggingface_model_id)
+    if summarization_type == "brief":
+        return gptExplainer.brief(github_url=github_project_url, branch=github_project_branch)["summary"]
+    return gptExplainer.outline(github_url=github_project_url, branch=github_project_branch)["summary"]
+
+
+import streamlit as st
```
Comment (kmehant) [inline project_explainer_ui/streamlit_ui.py:10]: Can we move imports to top cleanly?

File: project_explainer_ui/streamlit_ui.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from gh_explainer import Explainer
+
+def summarize(summarization_type, github_project_url, github_project_branch="main", huggingface_model_id="gpt2"):
```
Comment (kmehant) [inline project_explainer_ui/streamlit_ui.py:3]: Can you add a doc string?

**Review by kmehant — CHANGES_REQUESTED** [review]

File: project_explainer_ui/streamlit_ui.py (hunk 1 above)
Comment (debrupf2946) [reply project_explainer_ui/streamlit_ui.py:10]: Sorry, It was my mistake!,I will make it clean

**Comment by debrupf2946** [comment]
> [#19 (comment)](https://github.com/c2siorg/Project-Explainer/pull/19#issuecomment-1970799243)
> 
> Would you like to host it? Thank you.

Yes, I have already hosted it, but there is a [bug](https://github.com/c2siorg/Project-Explainer/pull/19/commits/98274b2ba23f1c3210df3f75699d2a0d61bcb871) which is causing an issue that's need to be resolved first I Think,[Description](https://github.com/c2siorg/Project-Explainer/commit/98274b2ba23f1c3210df3f75699d2a0d61bcb871#commitcomment-139230367)
[Deployed app](https://appuipy-j9rkwrbchxdgappbfkmxogf.streamlit.app/)
It might be dead now due to inactivity

**Pull request closed without merging**